alibi_detect/saving/loading.py

```diff
@@ -28,60 +28,38 @@
 
 logger = logging.getLogger(__name__)
 
-REQUIRES_BACKEND = [
-    'ClassifierDrift',
-    'ClassifierUncertaintyDrift',
-    'LearnedKernelDrift',
-    'LSDDDrift',
-    'MMDDrift',
-    'RegressorUncertaintyDrift',
-    'SpotTheDiffDrift'
-]
 
 # Fields to resolve in resolve_config ("resolve" meaning either load local artefact or resolve @registry, conversion to
 # tuple, np.ndarray and np.dtype are dealt with separately).
+# Note: For fields consisting of nested dicts, they must be listed in order from deepest to shallowest, so that the
+# deepest fields are resolved first. e.g. 'preprocess_fn.src' must be resolved before 'preprocess_fn'.
```

Comment:
However, `['preprocess_fn', 'src']` is in the opposite order?

Reply:
The way the config dict recursion works is such that:

```python
FIELDS_TO_RESOLVE = [
    ['preprocess_fn', 'src'],
    ['preprocess_fn', 'model'],
    ['preprocess_fn', 'embedding'],
    ['preprocess_fn', 'tokenizer'],
    ['preprocess_fn', 'preprocess_batch_fn'],
    ['preprocess_fn'],
```

means resolve `src` in `preprocess_fn`, then resolve `model` in `preprocess_fn`, etc etc, and finally resolve `preprocess_fn`. We need to have resolved all the possible artefacts in `preprocess_fn` before we resolve `preprocess_fn` itself. Hence why `['preprocess_fn', 'src']` must be listed before `['preprocess_fn']`. Don't know if that clarifies things?

Reply:
Alright, I was thinking resolving happens on every item in row, but actually it's row-by-row.